app/serializers/serialize_exercise_instructions.rb

```diff
@@ -0,0 +1,72 @@
+class SerializeExerciseInstructions
+  include Mandate
+
+  initialize_with :exercise
+
+  def call
+    {
+      overview: overview,
+      general_hints: general_hints,
+      tasks: tasks
+    }
+  end
+
+  private
+  def overview
+    instructions_doc.each.
+      take_while { |node| node.type != :header }.
+      map(&:to_html).
+      join
```

Comment:
This has a small hidden bug, as just concatenating the HTML results in spacing being lost. I have to check how the HTML would be rendered when the elements are combined.